tests/countries/test_niger.py

```diff
@@ -0,0 +1,377 @@
+        self.assertNoNonObservedHolidayName(name, obs_dt)
+        self.assertNoHolidayName(f"{name} (lura)", range(1959, 1998))
+
+    def test_ascension_thursday(self):
+        name = "Hawan Yesu zuwa sama Alhamis"
+        self.assertHolidayName(
+            name,
+            "2020-05-21",
+            "2021-05-13",
+            "2022-05-26",
+            "2023-05-18",
+            "2024-05-09",
+            "2025-05-29",
+        )
+        self.assertHolidayName(name, range(1959, 2050))
+
+    def test_whit_sunday(self):
+        name = "Fentikos"
+        self.assertHolidayName(
+            name,
+            "2020-05-31",
+            "2021-05-23",
+            "2022-06-05",
+            "2023-05-28",
+            "2024-05-19",
+            "2025-06-08",
+        )
+        self.assertHolidayName(name, range(1959, 2050))
+        self.assertNoHolidayName(f"{name} (lura)", range(1959, 1998))
+
+    def test_cnsp_coup_anniversary(self):
+        name = "Bikin bukin juyin mulkin CNSP"
+        self.assertHolidayName(
+            name,
+            "2024-07-26",
+            "2025-07-26",
+        )
+        self.assertHolidayName(name, (f"{year}-07-26" for year in range(2024, 2050)))
+        self.assertNoHolidayName(name, range(1959, 2024))
+
+    def test_independence_day(self):
+        name = "Bikin cikar shelar 'yancin kai"
+        self.assertHolidayName(
+            name,
+            "2020-08-03",
+            "2021-08-03",
+            "2022-08-03",
+            "2023-08-03",
+            "2024-08-03",
+            "2025-08-03",
+        )
+        self.assertHolidayName(name, (f"{year}-08-03" for year in range(1960, 2050)))
+        self.assertNoHolidayName(name, 1959)
+        obs_dt = (
+            "2008-08-04",
+            "2014-08-04",
+            "2025-08-04",
+        )
+        self.assertHolidayName(f"{name} (lura)", obs_dt)
+        self.assertNoNonObservedHolidayName(name, obs_dt)
+        self.assertNoHolidayName(f"{name} (lura)", range(1959, 1998))
+
+    def test_assumption_of_mary(self):
+        name = "Zaton Maryama"
+        self.assertHolidayName(
+            name,
+            "2020-08-15",
+            "2021-08-15",
+            "2022-08-15",
+            "2023-08-15",
+            "2024-08-15",
+            "2025-08-15",
+        )
+        self.assertHolidayName(name, (f"{year}-08-15" for year in range(1959, 2050)))
+        obs_dt = (
+            "2010-08-16",
+            "2021-08-16",
+        )
+        self.assertHolidayName(f"{name} (lura)", obs_dt)
+        self.assertNoNonObservedHolidayName(name, obs_dt)
+        self.assertNoHolidayName(f"{name} (lura)", range(1959, 1998))
+
+    def test_all_saints_day(self):
+        name = "Rana Duka Saints"
+        self.assertHolidayName(
+            name,
+            "2020-11-01",
+            "2021-11-01",
+            "2022-11-01",
+            "2023-11-01",
+            "2024-11-01",
+            "2025-11-01",
+        )
```

Comment:
Duplicated from `self.assertHolidayName(name, (f"{year}-11-01" for year in range(1959, 2050)))`
```suggestion
```